src/data_collectors.py

```diff
@@ -0,0 +1,71 @@
+import abc
+from concurrent.futures import Future
+from functools import partial
+import requests
+import re
+
+from constants import TIMEOUT_RSS_REQUEST
+from helper import get_request_with_timeout
+
+class BaseDataCollector(object): 
+    def __init__(self):
+        pass
+    
+    @abc.abstractmethod
+    def get_data_collection_futures(self, executor):
+        pass
+
+
+class RssDataCollector(BaseDataCollector): 
+    def __init__(self, base_url):
+        super().__init__()
+        self.base_url = base_url
+    
+
+    def get_data_collection_futures(self, executor):
+        feed_urls = self._get_feed_urls()
+        get_request = get_request_with_timeout(TIMEOUT_RSS_REQUEST)
+        futures = list(map(lambda url: executor.submit(get_request, url), feed_urls))
+        return futures
+
+    def _get_feed_urls(self):
+        headers = {
```

Comment:
do these headers need to be editable / configurable? if so, maybe move them to some maintainable configuration or .env file?